File: OCH_Tao/ABC/B/054.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const [N, M] = (input.shift().split(" ")).map(x => parseInt(x, 10));
let A = [];
let B = [];
for (let i = 0; i < N; i++) {
  A.push(input.shift());
}
for (let i = 0; i < M; i++) {
  B.push(input.shift());
}
for (let i = 0; i < N - M + 1; i++) {
  if (A[i].includes(B[0])) {
    let x = A[i].indexOf(B[0]);
    for (let j = 1; j < M; j++) {
      if (A[i + j].slice(x, x + M) !== B[j]) {
        break;
      }
    }
    console.log("Yes");
    break;
  }
  console.log("No");
  break;
}

File: OCH_Tao/ABC/B/054.py
N,M = map(int,input().split())
A = []
B = []
for i in range(N):
  A.append(input())
for i in range(M):
  B.append(input())
for i in range(N-M+1):
  if B[0] in A[i]:
    x = A[i].find(B[0])
    for j in range(1,M):
      if (A[i+j])[x:x+M] != B[j]:
        break
    else:
      print("Yes")
      break
else:
  print("No")